test/k8s-e2e/e2e-test-execute

```diff
@@ -171,45 +207,72 @@ function build_k8s() {
 
 	pushd "$K8S_ROOT/kubernetes"
 	echo -e "$(info INFO:) Get kubetest"
+	go get -u k8s.io/examples
 	go get -u k8s.io/test-infra/kubetest
 }
 
 
 function generate_description_file() {
 	# If e2e.log and junit_01.xml shall be published, the ginkgo.focus must be \[Conformance\]
-	if [[ "$publish_test_results" -eq "$TRUE" ]]; then
+	if [ "$publish_test_results" = "true" ]; then
 		generated_run_desc_path=$test_desc_path
 		return 0
 	fi
 
 	echo -e "$(info INFO:) Start generating $generated_run_desc_path"
 
-	if [[ "$run_all_failing_tests" -eq "$TRUE" ]]; then
+	if [ "$run_all_failing_tests" = "true" ]; then
+		# Remove tests from e2e_base_fast.desc
+		grep -Fxvf "$DESCRIPTIONS_DIR"/e2e_base_fast.desc "$all_tests_desc_path" > "$DESCRIPTIONS_DIR"/tmp1
 
-		# Remove tests from successful.desc
-		grep -Fxvf "$DESCRIPTIONS_DIR"/successful.desc "$all_tests_desc_path" > "$DESCRIPTIONS_DIR"/tmp
+		# Remove tests from e2e_base_slow.desc
+		grep -Fxvf "$DESCRIPTIONS_DIR"/e2e_base_slow.desc "$DESCRIPTIONS_DIR"/tmp1 > "$DESCRIPTIONS_DIR"/tmp2
 
 		# remove gke only tests
-		grep -Fxvf "$DESCRIPTIONS_DIR"/gke_tests.desc "$DESCRIPTIONS_DIR"/tmp > "$generated_run_desc_path"
-		rm -f "$DESCRIPTIONS_DIR"/tmp
+		grep -Fxvf "$DESCRIPTIONS_DIR"/gke_tests.desc "$DESCRIPTIONS_DIR"/tmp2 > "$DESCRIPTIONS_DIR"/tmp1
+
+		# remove blacklisted tests
+		grep -Fxvf "$DESCRIPTIONS_DIR"/blacklist.desc "$DESCRIPTIONS_DIR"/tmp1 > "$generated_run_desc_path"
+
+		# clean up temporary files
+		rm -f "$DESCRIPTIONS_DIR"/tmp1
+		rm -f "$DESCRIPTIONS_DIR"/tmp2
 
 		# remove conformance tests
 		sed -i '/\[Conformance\]/d' "$generated_run_desc_path"
 	else
+		if grep '^\[Conformance\]$' $test_desc_path ; then
+			grep "\[Conformance\]" $all_tests_desc_path >> "$generated_run_desc_path"
+		fi
+
+		if [ ! $CONFORMANCE_FOCUS = "" ]; then
+			echo -e "$(info INFO:) keep tests that match: $CONFORMANCE_FOCUS"
+			sed -i '/'${CONFORMANCE_FOCUS}'/!d' "$generated_run_desc_path"
+		fi
+		if [ ! $CONFORMANCE_SKIP = "" ]; then
+			echo -e "$(info INFO:) skip tests that match: $CONFORMANCE_SKIP"
+			sed -i '/'${CONFORMANCE_SKIP}'/d' "$generated_run_desc_path"
+		fi
 		while read -r e2e_test; do
-			cat $all_tests_desc_path | grep "$(escape_test_name "${e2e_test}")" >> "$generated_run_desc_path"
+			if [ "[Conformance]" = "$e2e_test" ] ; then
+				continue
+			fi
+			grep "$(escape_test_name "${e2e_test}")" $all_tests_desc_path >> "$generated_run_desc_path"
 		done <<<"$(cat "${test_desc_path}")"
-	fi
 
-	if [[ $SERIAL_TESTS_ONLY -eq $TRUE ]]; then
-		sed -i '/\[Serial\]/!d' "$generated_run_desc_path"
-	elif [[ $SERIAL_TESTS_ONLY -eq $FALSE ]]; then
-		sed -i '/\[Serial\]/d' "$generated_run_desc_path"
+		# remove duplicates
+		sort -r -u -o "$generated_run_desc_path" "$generated_run_desc_path"
```

Comment:
why can there be duplicates?

Reply:
Sometimes testcase names only differ in suffix parts. E.g. testcase1=foo and testcase2=foobar. In this case matching for one test case may result in multiple matches if we look for testcases foo and foobar.